code/drasil-example/template/lib/Drasil/Template/Body.hs

```diff
@@ -141,3 +181,191 @@ progName = commonIdeaWithDict "progName" (pn "ProgName") "ProgName" []
 -- MOVE TO DATA.PEOPLE
 authorName :: Person
 authorName = person "Author" "Name"
+
+{-------------------------------------------------------------------------------
+- Equilibrium
+-------------------------------------------------------------------------------}
+
+equilibrium :: IdeaDict
+equilibrium = nc "equilibrium" $ nounPhrase "equilibrium" "equilibria"
+
+-- There is a generic copy between economics and physics, but we won't bother
+-- with that.
+
+{-------------------------------------------------------------------------------
+- Equilibrium (Economics)
+-------------------------------------------------------------------------------}
+
+equilibriumQ :: QuantityDict
+equilibriumQ = mkQuant' -- A somewhat odd variant of `mkQuant` that re-orders argument and adds staging to the `Symbol`
+  "inEquilibrium"
+  (nounPhrase "in equilibrium" "all equilibria") -- duplicates the IdeaDict
+  Nothing -- UnitDefn
+  Boolean
+  (autoStage $ label "inEquilibrium")
+  Nothing -- "abbreviation String"
+
+price :: IdeaDict -- I guess making this is completely unnecessary because the QuantityDict will make its own.
+price = nc "price" $ cn "price"
+
+priceQ :: QuantityDict
+priceQ = mkQuant'
+  "priceQuant" -- duplicates the IdeaDict
+  (cn "price")
+  Nothing -- UnitDefn // FIXME: Uh oh! There's seemingly no way we can define a new unit: $CAD.s
+  Real -- Not quite a "Real," more of a "dollar amount with two decimal places, or a tuple containing two integers, with the second capped at 100"
+  (autoStage cP)
+  Nothing -- Abbreviation
+
+supply :: IdeaDict
+supply = nc "supply" $ cnIES "supply"
+
+supplyDemandFuncSpace :: Space
+supplyDemandFuncSpace = mkFunction [Real] Integer -- Real should be Dollar, output should be strictly non-negative integers (Z^+)
```

Comment:
Strictly non-negative integers (as well as natural numbers) would be great additions to our `Space` system!